### DISCUSSION TITLE
Reverb random deactivation

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1231] [Help]
I have a small probelm in GrandOrgue: the reverb feature is randomly disabled. This can happen at startup, after a few minutes or 2 hours of play. Either way, I just open the Settings menu and click "Ok" to turn the reverb feature back on. I don't even need to go to the reverb tab. If I do anyway, I see that the box is checked.

My configuration : Windows 10 / GrandOrgue v3.7.0-1. I also have GrandOrgue installed on a Raspberry Pi but I never experienced this issue.

**Comment by larspalo** [comment]
Could you please provide some additional information? (Something like a confirmation that you actually added the impulse response file yourself _and_ enabled the feature) The **only** additional reverb feature (besides what reverb is actually recorded in the samples themselves) inside of GO is impulse response (IR) reverb that you add in the appropriate tab in the settings dialog (with an appropriate IR file). Are you saying that the activated IR feature (and loaded IR file) is actually randomly turned off?

**Comment by newVideophile** [comment]
The short answer is yes to all your questions. I use the IR feature with my own previously loaded file. The random deactivation will typically occur 1 to 3 times in a 2 hour playing session.

**Comment by newVideophile** [comment]
I recently (July 30) updgraded my Raspberry Pi with the latest version of Raspberry Pi OS and GrandOrgue and I encountered this problem, but surprisingly only in one ear ! That is, the reverb was deactivated in the right channel, but still active in the left. The problem occurs randomly after a certain amount of playing time.

The Raspberry Pi worked fine for over 2 years, until I tried to updgrade. 

Since two very different machines (different hardware, different OS, different audio interface) have the same type of problem, I think it's worth investigating. I can share my impulse response file, if that helps.

**Comment by newVideophile** [comment]
Is there a developer around willing to investigate this issue?

Am I the only user with this bug?

**Comment by newVideophile** [comment]
Is there anyone ? ONE... One... one...

**Reply by larspalo** [reply]
It's very hard to investigate anything when you (read in this case: I) cannot reproduce it. I've never experienced this myself even after many hours of leaving GO on.

**Reply by newVideophile** [reply]
Well, yes, but at least it could be possible to review the code that handles the reverb...

**Reply by newVideophile** [reply]
I am willing to try if you can guide me in the source files. I am a programmer myself, although more familiar with microcontrollers and embedded systems

**Reply by rousseldenis** [reply]
This makes me think of the issue I had. I changed the polyphony setting and it solved it. I don't remember if it was decreasing or increasing the value.

What I found was about the CPU consumption when playing fast polyphony songs like Mendhelsson's sonates.

**Comment by newVideophile** [comment]
Ah I feel less alone with my problem!

In my experience, opening and closing the setting menu would cure the problem, even without changing anything.

I have not found any triggering condition. CPU consumption may be one, but I did not pay attention to that. I will test that again when I have some time (for the moment, I prefer using my scarce free time as an organist rather than an engineer)

**Comment by oleg68** [comment]
> I am willing to try if you can guide me in the source files: 


Unfortunaly, for reveberation GrandOrgue uses an external source codes of ``Zita Convolver`` from here: https://salsa.debian.org/multimedia-team/zita-convolver, and nobody of the GrandOrgue developers is familiar with them.

So several versions ago there was a warning that the reveberation feature was experimental and it was not supported.
